check_exec.go

```diff
@@ -26,10 +27,10 @@ func (c CheckExec) Validate() error {
 	return nil
 }
 
-func (c CheckExec) Execute(timeout time.Duration) error {
+func (c CheckExec) Execute(timeout time.Duration) (bool, error) {
```

Comment:
Rather than an additional return argument, I'd be tempted to implement a new error type (like http://play.golang.org/p/odS5eBBXjW) and create a wrapper around Execute which interprets said error type, this decoupling the execution and error handling stage.